### PR TITLE
Added listGists method

### LINKED ISSUE #1325: Accessing Gists via getMyself().listGists() returns null data for Gist files
Once I authenticate to GitHub with a valid access token, but then attempt to pull the Map of Gists using this method:

`gitHub.getMyself().listGists();`

It will give me a map of GHGist objects and their gistId's, but when I then use the GHGist object to pull the files out of Gist, the file data comes back null.

The only way I am able to access files successfully, is if I first iterate through that map and build a list of Gist Ids like this:

```
 List<String> gistIDs = new ArrayList<>();
 for (GHGist gist : gitHub.getMyself().listGists()) {
     String gistID = gist.getGistId();
     gistIDs.add(gistID);
 }
```

Then iterate through that list and specifically call for the Gist object from the parent connection like this:

```
for (String gistID : gistIDs) {
    GHGist gist = gitHub.getGist(gistID);
    for (String filename : gist.getFiles().keySet()) {
        GHGistFile file = gist.getFiles().get(filename);
        System.out.println("Contents: " + file.getContent());
    }
}

```
ONLY by using this two-step process, can I access the files in the Gist. I believe this is a bug in the API.

Thank you,

Mike

### PR DESCRIPTION
**Pull request by EasyG0ing1**
# Description

Currently, when pulling down a list of GHGist objects, using
```Java
gitHub.getMyself().listGists().toList()
```
The GHGist objects returned in the list do not contain the contents of the files within each Gist, and there is no other method from which to pull down all gists with complete files in a single list. The reason for this seems to be that the GitHub API URL path that is used to pull the list from, intentionally does not include the contents of the file. The API docs seem to indicate a conservative data yield for that particular path, requiring a developer to then use another URL path from which to pull the individual Gists, which would then include the contents of the file.

I added this method to the GitHub class, which I have tested and verified that the list it returns is a complete list of the user's GHGist objects, where each file within the Gists has all of the data associated with the file, including content.

Fixes #1325 

The only issue that came from ```mvn -D enable-ci clean install site``` was a warning about inconsistent synchronization, but it didn't just flag my code, it flagged other code in the class as well. It also indicates that the code might be just fine and that it could be a false positive due to the nature of the test. 

The method is relatively simple, I cannot see where it would cause any issues. However, I'm willing to make whatever changes are necessary.

# Before submitting a PR:

- [X] Changes must not break binary backwards compatibility. If you are unclear on how to make the change you think is needed while maintaining backward compatibility, [CONTRIBUTING.md](CONTRIBUTING.md) for details.
- [X] Add JavaDocs and other comments as appropriate. Consider including links in comments to relevant documentation on https://docs.github.com/en/rest .  
- [ ] Add tests that cover any added or changed code. This generally requires capturing snapshot test data. See [CONTRIBUTING.md](CONTRIBUTING.md) for details.
- [X] Run `mvn -D enable-ci clean install site` locally. If this command doesn't succeed, your change will not pass CI.
- [X] Push your changes to a branch other than `main`. You will create your PR from that branch.

# When creating a PR: 

- [X] Fill in the "Description" above with clear summary of the changes. This includes:
  - [X] If this PR fixes one or more issues, include "Fixes #<issue number>" lines for each issue. 
  - [ ] Provide links to relevant documentation on https://docs.github.com/en/rest where possible.
- [ ] All lines of new code should be covered by tests as reported by code coverage. Any lines that are not covered must have PR comments explaining why they cannot be covered. For example, "Reaching this particular exception is hard and is not a particular common scenario."
- [X] Enable "Allow edits from maintainers".

### CODE REVIEW

File: src/main/java/org/kohsuke/github/GitHub.java [hunk 1]
```diff
@@ -919,6 +920,22 @@ public GHGist getGist(String id) throws IOException {
...
+     * Gets a List of all GHGist objects
+     *
+     * @return the list
+     * @throws IOException
+     *             the io exception
+     */
+    public List<GHGist> listGists() throws IOException {
+        CopyOnWriteArrayList<GHGist> list = new CopyOnWriteArrayList<>();
+        assert myself != null;
+        for (GHGist ghGist : myself.listGists().toList()) {
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GitHub.java:933]:
```suggestion
        for (GHGist ghGist : getMyself().listGists().toList()) {
```

**Review by bitwiseman — CHANGES_REQUESTED** [review]
Please add at least one test for your new method.

**Comment by EasyG0ing1** [comment]
> Please add at least one test for your new method.

The test will need to use an account that has Gists and files in the Gists in order to work. I wrote a test that authenticates with my own token and it works, but I don't know how to write the test so that it will always work regardless of who is running it ... since it needs a valid account that has Gists in it.

**Comment by bitwiseman** [comment]
@EasyG0ing1 

You can record a snapshot of the tests and then add the `snapshotNotAllowed();` method to prevent it from being overwritten later. 

However, I'm going o make another suggestion.  Instead of adding a new method to get the list Gists, I suggest you add a `populate()` method to `GHGist` that calls the URL that returns the full contents similar to other `populate()` methods:

https://github.com/hub4j/github-api/blob/bcb71a3c318d243c92b757310cf153b8afc983d5/src/main/java/org/kohsuke/github/GHLicense.java#L196-L19

That will allow people to get the file contents if needed while preserving the incremental loading.   The test will also be simpler. You can add a step at the end of the existing test for listing Gists.

**Pull request closed without merging**